src/include/homestore/replication/repl_dev.h

```diff
@@ -98,10 +130,10 @@ class ReplDev {
     /// cases
     /// @param value - vector of io buffers that contain value for the key. It is an optional field and if the value
     /// list size is 0, then only key is written to replicadev without data.
-    /// @param user_ctx - User supplied opaque context which will be passed to listener
+    /// @param ctx - User supplied context which will be passed to listener
     /// callbacks
     virtual void async_alloc_write(sisl::blob const& header, sisl::blob const& key, sisl::sg_list const& value,
-                                   void* user_ctx) = 0;
+                                   intrusive< repl_req_ctx > ctx) = 0;
```

Comment:
do you mind add a interface: get_blk_size() to ReplicationService,  because value size must be multiples of blk size , but for upper layer(HO) does not what is blk size at all. HS datasvc has such a interface, but it will be strange to call datasvc get_blk_size, it will be better to call replsvc get_blk_size().
it is a small change and I am both ok to add it in this pr or use another pr.

Reply:
Make sense, will add as part of this PR itself.